api.h

```diff
@@ -0,0 +1,64 @@
+#ifndef API_H
+#define API_H
+
+/*************************************************
+ * Name:        XMSS_crypto_sign_keypair
+ *
+ * Description: Generates public and private key.
+ *
+ * Arguments:   - uint8_t *pk: pointer to output public key (allocated
+ *                             array of CRYPTO_PUBLIC_KEY bytes)
+ *              - uint8_t *sk: pointer to output private key (allocated
+ *                             array of CRYPTO_SECRET_KEY bytes)
+ *
+ * Returns 0 (success), -1 otherwise
+ **************************************************/
+int crypto_sign_keypair(unsigned char *pk, unsigned char *sk);
+
+/*************************************************
+ * Name:        XMSS_crypto_sign
+ *
+ * Description: Computes signature.
+ *
+ * Arguments:   - uint8_t *sm:   pointer to output signature (of length CRYPTO_BYTES)
+ *              - uint8_t *m:     pointer to message to be signed
+ *              - uint8_t *sk:    pointer to bit-packed secret key
+ *              - unsigned long long *smlen: pointer to output length of signature
+ *              - unsigned long long mlen:    length of message
+ *
+ * Returns 0 (success), -1 otherwise
+ **************************************************/
+int crypto_sign(unsigned char *sm, unsigned long long *smlen,
```

Comment:
Is the output array left unchanged on error?

Reply:
On error, the output arrays probably are modified. But I expect the caller will check error return before using those array.